unittest/freedv_700d_comptx.c

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------*\
+
+  freedv_comptx.c
+
+  Complex valued Tx to support ctests.
+
+\*---------------------------------------------------------------------------*/
+
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "freedv_api.h"
+
+int main(int argc, char *argv[]) {
+    struct freedv            *freedv;
+
+    freedv = freedv_open(FREEDV_MODE_700D);
+    assert(freedv != NULL);
+
+    /* handy functions to set buffer sizes */
+    int n_speech_samples = freedv_get_n_speech_samples(freedv);
+    short speech_in[n_speech_samples];
+    int n_nom_modem_samples = freedv_get_n_nom_modem_samples(freedv);
+    COMP mod_out[n_nom_modem_samples];
+    short mod_out_short[2*n_nom_modem_samples];
+
+    /* OK main loop  --------------------------------------- */
+
+    while(fread(speech_in, sizeof(short), n_speech_samples, stdin) == n_speech_samples) {
+        freedv_comptx(freedv, mod_out, speech_in);
+        for(int i=0; i<n_nom_modem_samples; i++) {
+            mod_out_short[2*i] = mod_out[i].real;
```

Comment:
Are there any architectures where COMP isn't going to work out to two floats next to each other? If not, could this be simplified such that mod_out is written directly?

Reply:
Breaking the API and using knowledge about internal data structures is not a good idea.  We use `COMP` for good reason - to make it clear this is a complex valued number.

Reply:
Fair enough 👍